lib/kino/process.ex

```diff
@@ -0,0 +1,229 @@
+  """
+
+  alias Kino.Markdown
+
+  @type supervisor :: pid() | atom()
+
+  @doc """
+  Generates a visualization of a supervision tree.
+
+  The provided supervisor can be either a named process or a PID. The supervision tree
+  is displayed with solid lines denoting supervisor-worker relationships and dotted
+  lines denoting links between processes. The graph rendering supports the following
+  options:
+
+  ## Options
+
+    * `:direction` - defines the direction of the graph visual. The
+      value can either be `:top_down` or `:left_right`. Defaults to `:top_down`.
+
+  ## Examples
+
+  With a supervisor definition like so:
+
+      {:ok, supervisor_pid} =
+        Supervisor.start_link(
+          [
+            {DynamicSupervisor, strategy: :one_for_one, name: MyApp.DynamicSupervisor},
+            {Agent, fn -> [] end}
+          ],
+          strategy: :one_for_one,
+          name: MyApp.Supervisor
+        )
+
+      Enum.each(1..3, fn _ ->
+        DynamicSupervisor.start_child(MyApp.DynamicSupervisor, {Agent, fn -> %{} end})
+      end)
+
+  You can then call `Kino.Process.sup_tree/1` to render
+  the supervision tree using using the PID of the supervisor.
+
+      Kino.Process.sup_tree(supervisor_pid)
+
+  You can also render the supervisor by passing the name of the supervisor
+  if the supervisor was started with a name.
+
+      Kino.Process.sup_tree(MyApp.Supervisor)
+
+  You can also change the direction of the rendering by calling
+  `Kino.Process.sup_tree/2` with the `:direction` option.
+
+      Kino.Process.sup_tree(MyApp.Supervisor, direction: :left_right)
+  """
+  @spec sup_tree(supervisor(), keyword()) :: Markdown.t()
+  def sup_tree(supervisor, opts \\ []) do
+    supervisor =
+      case GenServer.whereis(supervisor) do
+        supervisor_pid when is_pid(supervisor_pid) ->
+          supervisor_pid
+
+        _ ->
+          raise ArgumentError,
+                "the provided identifier #{inspect(supervisor)} does not reference a running supervisor"
+      end
+
+    direction =
+      opts
+      |> Keyword.get(:direction, :top_down)
+      |> convert_direction()
+
+    try do
+      initial_pid_lookup = Map.put(%{}, supervisor, {0, :supervisor})
+
+      edges =
+        supervisor
+        |> Supervisor.which_children()
+        |> traverse_processes(supervisor, {%{}, 1, initial_pid_lookup})
+        |> traverse_links()
+
+      edges =
+        edges
+        |> Enum.map_join("\n", fn {_pid_pair, edge} ->
+          generate_mermaid_entry(edge)
+        end)
+
+      Kino.Markdown.new("""
+      ```mermaid
+      graph #{direction};
+      #{edges}
+      classDef root fill:#c4b5fd, stroke:#374151, stroke-width:4px;
+      classDef supervisor fill:#c4b5fd, stroke:#374151, stroke-width:1px;
+      classDef worker fill:#93c5fd, stroke:#374151, stroke-width:1px;
+      ```
+      """)
+    catch
+      _, _ ->
+        raise ArgumentError, "the provided PID #{inspect(supervisor)} is not a supervisor"
```

Comment:
Can we wrap the `Supervisor.which_children` call specifically? Otherwise it's not obvious where this comes from :)